api/Controllers/FacilitiesController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using iTechArt.Hotels.Api.Entities;
+using iTechArt.Hotels.Api.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using System.Threading.Tasks;
+using static iTechArt.Hotels.Api.Constants;
+
+namespace iTechArt.Hotels.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class FacilitiesController : Controller
+    {
+        private readonly HotelsDatabaseContext _hotelsDb;
+        private readonly IMapper _mapper;
+
+        public FacilitiesController(HotelsDatabaseContext hotelDb, IMapper mapper)
+        {
+            _hotelsDb = hotelDb;
+            _mapper = mapper;
+        }
+
+        [HttpGet]
+        [Authorize(Roles = Role.Admin)]
+        public async Task<Facility[]> GetFacilities([FromQuery] FacilityParams facilityParams)
+        {
+            if (facilityParams.RoomId != null)
+            {
+                return await _hotelsDb.Facilities
+                    .Where(facility => facility.Realm == Realm.Room)
+                    .ProjectTo<Facility>(_mapper.ConfigurationProvider)
+                    .ToArrayAsync();
+
+            }
+            if (facilityParams.HotelId != null)
+            {
+                return await _hotelsDb.Facilities
+                    .Where(facility => facility.Realm == Realm.Hotel)
+                    .ProjectTo<Facility>(_mapper.ConfigurationProvider)
+                    .ToArrayAsync();
+            }
+            return await _hotelsDb.Facilities
+                .ProjectTo<Facility>(_mapper.ConfigurationProvider)
+                .ToArrayAsync();
+        }
```

Comment:
Why did you decide to make a single method instead of 3?

Reply:
GetFacilities method used to look really confusing before it was rewritten, so I took out it's logic into two other methods. Now It looks much better and I don't think I should be divided into other methods